### PR TITLE
[Macrobenchmark] Add a basic Compose LazyList sample

### PR DESCRIPTION
**Pull request by bentrengrove**
The goal was to match the current RecyclerView sample as close as possible. This will allow us to compare the results.

### Results

Startup
```
SmallListStartupBenchmark_startup[mode=COLD]
timeToInitialDisplayMs   min 159.9,   median 168.2,   max 187.0
Traces: Iteration 0 1 2 3 4
SmallListStartupBenchmark_startupCompose[mode=COLD]
timeToInitialDisplayMs   min 163.1,   median 170.6,   max 178.1
Traces: Iteration 0 1 2 3 4
SmallListStartupBenchmark_startup[mode=WARM]
timeToInitialDisplayMs   min 78.0,   median 89.6,   max 96.7
Traces: Iteration 0 1 2 3 4
SmallListStartupBenchmark_startupCompose[mode=WARM]
timeToInitialDisplayMs   min  82.2,   median  93.9,   max 100.5
Traces: Iteration 0 1 2 3 4
```

Scroll
```
FrameTimingBenchmark_scrollList
frameCpuTimeMs   P50   2.5,   P90   3.3,   P95  10.2,   P99  16.0
Traces: Iteration 0 1 2 3 4 5 6 7 8 9
FrameTimingBenchmark_scrollComposeList
frameCpuTimeMs   P50   2.9,   P90   5.7,   P95  10.4,   P99  12.3
Traces: Iteration 0 1 2 3 4 5 6 7 8 9
```

### CODE REVIEW

File: MacrobenchmarkSample/app/build.gradle [hunk 1]
```diff
@@ -23,7 +23,7 @@ android {
     compileSdkVersion 31
     defaultConfig {
         applicationId "com.example.macrobenchmark.target"
-        minSdkVersion 14
+        minSdkVersion 21
```
Comment (bentrengrove) [inline MacrobenchmarkSample/app/build.gradle:26]: Bumped this to 21 for Compose, let me know if I need to workaround the original value

Comment (tikurahul) [reply MacrobenchmarkSample/app/build.gradle:26]: No, `21` should be okay.

File: MacrobenchmarkSample/macrobenchmark/src/main/java/com/example/macrobenchmark/frames/FrameTimingBenchmark.kt [hunk 2]
```diff
@@ -66,4 +68,31 @@ class FrameTimingBenchmark {
...
+            startupMode = StartupMode.WARM, // restarts activity each iteration
+            iterations = ITERATIONS,
+            // [END_EXCLUDE]
+            setupBlock = {
+                // Before starting to measure, navigate to the UI to be measured
+                val intent = Intent("$packageName.COMPOSE_ACTIVITY")
+                startActivityAndWait(intent)
+            }
+        ) {
+            val column = device.findObject(By.desc("MyLazyColumn"))
```
Comment (tikurahul) [inline MacrobenchmarkSample/macrobenchmark/src/main/java/com/example/macrobenchmark/frames/FrameTimingBenchmark.kt:89]: Nit: Give the `LazyColumn` a more meaningful description.

**Review by tikurahul — APPROVED** [review]

**Review by mlykotom — APPROVED** [review]

File: MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ComposeActivity.kt [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        setContent {
+            MaterialTheme {
+                LazyColumn(
+                    Modifier
+                        .fillMaxSize()
+                        .semantics {
+                            // There are no view ids in Compose so we use the content description
+                            // in order to find the Composable.
+                            contentDescription = LAZY_COLUMN_TAG
```
Comment (JolandaVerhoef) [inline MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ComposeActivity.kt:43]: Please don't use semantics properties to do anything else than conveying semantics information. In this case, you might confuse Accessibility Services like Talkback by setting the contentDescription. Instead - what about using the `testTag` or using a more generic matcher?

File: MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ComposeActivity.kt [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                        EntryRow(entry = item, Modifier.padding(8.dp))
+                    }
+                }
+            }
+        }
+    }
+
+    private fun entries(size: Int) = List(size) {
+        Entry("Item $it")
+    }
```
Comment (JolandaVerhoef) [inline MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ComposeActivity.kt:56]: Do you really need a private fun for this? Can you just move it inline?

File: MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ComposeActivity.kt [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            Text(
+                text = entry.contents,
+                modifier = Modifier
+                    .padding(16.dp)
+                    .wrapContentSize()
+            )
+
+            Spacer(modifier = Modifier.weight(1f))
+
+            Checkbox(
```
Comment (JolandaVerhoef) [inline MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ComposeActivity.kt:77]: When you use Checkbox from Compose 1.1 and up, it should automatically make it 48x48dp. Do you still need to set the padding?

File: MacrobenchmarkSample/build.gradle [hunk 6]
```diff
@@ -17,7 +17,8 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
     ext.kotlin_version = "1.6.10"
-    ext.benchmark_version = "1.1.0-beta01"
+    ext.benchmark_version = "1.1.0-beta02"
+    ext.compose_version = "1.1.0-rc03"
```
Comment (JolandaVerhoef) [inline MacrobenchmarkSample/build.gradle:21]: You can use stable now right?

File: MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ComposeActivity.kt (hunk 4 above)
Comment (bentrengrove) [reply MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ComposeActivity.kt:56]: I am matching the existing sample for consistency here so will leave this one

File: MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ComposeActivity.kt (hunk 5 above)
Comment (bentrengrove) [reply MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ComposeActivity.kt:77]: Yes it seems to, it matches the recyclerview implementation.

**Pull request merged**